core-bundle/src/Routing/Content/PageUrlResolver.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Content;
+
+use Contao\PageModel;
+use Symfony\Component\Routing\Route;
+
+class PageUrlResolver implements ContentUrlResolverInterface
```

Comment:
There can/should only be one resolver for each content type. Be aware that resolvers are not the same as providers. Be aware that the page providers are asked **after** the resolver returns a route. This can be a bit confusing: a resolver does not know about page types (doesn't know that `NewsModel` has a `{year}` parameter), the resolver is responsible for handling that.
https://github.com/contao/contao/pull/1516/files#diff-298334abb66c7cdc0fbe073b8eb7e131R125-R136